src/Storages/AlterCommands.cpp

```diff
@@ -937,14 +937,26 @@ void AlterCommands::apply(StorageInMemoryMetadata & metadata, ContextPtr context
     {
         try
         {
-            new_projections.add(ProjectionDescription::getProjectionFromAST(projection.definition_ast, metadata_copy.columns, context));
+            if (projection.name != ProjectionDescription::VIRTUAL_PROJECTION_NAME)
```

Comment:
What is the case for such an exception? When name for user-defined projection is the same as the name for a virtual?

Is it a good idea to store virtual projections in ProjectionsDescription, in table metadata?
I think we can assume that such a projection just always exists for MergeTree at reading time. Probably we can make then unnamed - or just don't care about name clash.
Looks like having it in metadata only cause more complications.

Reply:
Sorry, it should have been a comment to line 955